src/routes/write-exercise/WriteExerciseScreen.tsx

```diff
@@ -101,6 +105,22 @@ const WriteExerciseScreen = ({ route, navigation }: WriteExerciseScreenProps): R
     setIsAnswerSubmitted(true)
   }
 
+  useEffect(() => {
+    if (hasCheated) {
+      needsToBeRepeated = false
+      continueExercise().catch(log)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [documentsWithResults])
+
+  const cheatExercise = async (result: SimpleResult): Promise<void> => {
+    setHasCheated(true)
+    setCurrentIndex(documentsWithResults.length - 1)
+
+    const cheatedDocuments = documentsWithResults.map(it => ({ ...it, numberOfTries: 1, result })).reverse()
```

Comment:
Any reason why you are using `reverse`?

Reply:
Forgot to remove, sorry. Was running into an issue where the useEffect hook didn't fire, since documentsWithResults don't change when failing all.